### DISCUSSION TITLE
"keep_blank_chars" not supported in "extract_table" for version 0.8.1

### DISCUSSION THREAD

**Discussion by malek-ewa** [jsvine/pdfplumber#861] [Q&A] [answered]
Hello there 👋

I had a code that's been working fine so far, it's extracting tables from a PDF, and it looks like this:

```
page.extract_table(table_settings={
    "keep_blank_chars" : True,
    "horizontal_strategy": "text",
    "snap_y_tolerance": 4,
    "vertical_strategy": "lines",
    "intersection_x_tolerance": 100,
    "explicit_horizontal_lines": page.curves+page.edges
})
```

The code was working fine with version `0.7.4` of pdfplumber, until I upgraded to version `0.8.1`, and then it started throwing a  `TypeError: __init__() got an unexpected keyword argument 'keep_blank_chars'` error for the above code.

Is this an expected behaviour ? Was `keep_blank_chars` removed in the most recent versions ?

Thanks in advance.
Malek

**Comment by jsvine** [comment, marked as answer]
Hi @malek-ewa, and my apologies for the inconvenience. In [v0.8.0](https://github.com/jsvine/pdfplumber/blob/stable/CHANGELOG.md#080---2023-02-13), `keep_blank_chars` was changed to `text_keep_blank_chars` to be more consistent with the rest of the text-based options, and to make it easier for `pdfplumber` to relay those options to the text extraction methods.

**Reply by feelosophy13** [reply]
@jsvine FYI, the README still references the `keep_blank_chars` instead of `text_keep_blank_chars`.

**Reply by jsvine** [reply]
Apologies for the confusion. This was fixed in https://github.com/jsvine/pdfplumber/commit/401b82cf747abd0b0fb2daac88d2d2813cd479e7 and will be in the `stable`-branch README in the next release, hopefully soon.
